test/server/listener_manager_impl_test.cc

```diff
@@ -3203,6 +3203,8 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, TransparentFreebindListenerDisabl
     name: "TestListener"
     address:
       socket_address: { address: 127.0.0.1, port_value: 1111 }
+    transparent: false
```

Comment:
Does this test fail without the changes you made? Mainly trying to understand if we are actually testing the change?

Reply:
yes, the test will fail if you remove the change in listener_impl.cc. Because `transparent: false`  causes config.has_transparent() return true, `EXPECT_EQ(options, nullptr);` fails this case.

Reply:
Yup OK, awesome, thanks.